py/desispec/sky.py

```diff
@@ -1036,142 +1040,167 @@ def subtract_sky(frame, skymodel, throughput_correction = False) :
-                
-                mcoef=np.sum(ivar*coef)/a                    
-                mcoeferr=1/np.sqrt(a)                
-                
-                nsig=3.
-                chi2=ivar_for_outliers*(coef-mcoef)**2                
-                worst=np.argmax(chi2)
-                if chi2[worst]>nsig**2*np.median(chi2[chi2>0]) : # with rough scaling of errors
-                    #log.debug("discard a bad measurement for fiber %d"%(fiber))
-                    ivar[worst]=0
-                    ivar_for_outliers[worst]=0
-                else :                    
-                    break
-
-            if failed :
-                continue
-                
-            log.info("fiber #%03d throughput corr = %5.4f +- %5.4f (mean fiber flux=%f)"%(fiber,mcoef,mcoeferr,np.median(frame.flux[fiber])))
-            
-            '''
-            if np.abs(mcoef)>0.01 :
-                
-                print(fiber,"mean coef=",mcoef,"all coef=",coef)
-                print(fiber,"all err=",np.sqrt(var))
-                print(fiber,"mean coef=",mcoef,"selected coef=",coef[ivar>0])
-                print(fiber,"select err=",np.sqrt(var[ivar>0]))
-                import matplotlib.pyplot as plt
-                x=np.arange(coef.size)
-                plt.errorbar(x,coef,np.sqrt(var),fmt="o")
-                plt.errorbar(x[ivar>0],coef[ivar>0],np.sqrt(var[ivar>0]),fmt="o")
-                plt.axhline(0.)
-                plt.axhline(mcoef)
-                plt.ylim(-0.11,0.11)
-                plt.grid()
-                plt.show()
-            '''
-            
-            if mcoeferr>np.abs(mcoef-1) :
-                log.warning("throughput corr error = %5.4f is too large compared to the correction value = %5.4f for fiber #%03d, do not apply correction"%(mcoeferr,(mcoef-1),fiber))
-                throughput_correction_value = 1.0
-            else :
-                throughput_correction_value = mcoef
-        
             # apply this correction to the sky model even if we have not fit it (default can be 1 or 0)
-            skymodel.flux[fiber] *= throughput_correction_value
+            skymodel.flux[fiber] *= skymodel.throughput_corrections[fiber]
     
     frame.flux -= skymodel.flux
     frame.ivar = util.combine_ivar(frame.ivar, skymodel.ivar)
     frame.mask |= skymodel.mask
 
     log.info("done")
 
+def calculate_throughput_corrections(frame,skymodel):
+    """
+    Calculate the throughput corrections for each fiber based on the skymodel.
+
+    Args:
+        frame (Frame object): frame containing the data that may need to be corrected
+        skymodel (SkyModel object): skymodel object that contains the information about the sky for the given exposure/frame
+
+    Output:
+        corrections (dict):  keys are the FIBER number and the values are the multiplicative corrections that would 
+                             be applied to the fluxes in frame.flux to correct them based on the input skymodel
```

Comment:
docstring says this returns a dict keyed by fiber, but it looks like it is returning an array indexed by fiber%500.

Reply:
This was left over from my first incarnation of the function. I've corrected this